### DISCUSSION TITLE
Uploading CSV file and ask questions about it

### DISCUSSION THREAD

**Discussion by htappe** [sashirestela/simple-openai#220] [Q&A] [answered]
After having looked at the examples I am still not quite sure how to start a thread where I upload a CSV file (preferably from stream) and ask questions about it. Also I would like to let the AI create some charts from the CSV data. Can anybody help?

**Comment by sashirestela** [comment, marked as answer]
Hi @htappe thanks for using simple-openai!

I think your question is not related to simple-openai functionalities but OpenAI API capabilities. Simple-OpenAI allows you to use those capabilities in Java. According to OpenAI documentation, it doesn't support retrievals over structured file formats (like csv or jsonl) yet. Take a look [HERE](https://platform.openai.com/docs/assistants/tools/file-search#how-it-works) at the Known Limitations section.

As an alternative, you could develop your own custom RAG system.
